src/main/java/org/jabref/gui/util/CustomFilteredList.java

```diff
@@ -0,0 +1,321 @@
+ * <ol>
+ * <li>Offers access to the {@link #refilter()} method, to re-evaluation of the filter predicate over the entire list.</li>
+ * <li>Adds the {@link #refilter(int, int)} method to re-evaluate the filter predicate over a specific range of elements.</li>
+ * </ol>
+ *
+ */
+
+public final class CustomFilteredList<E> extends TransformationList<E, E> {
+    private int[] filtered;
+    private int size;
+    private SortHelper helper;
+    private ObjectProperty<Predicate<? super E>> predicate;
+
+    public CustomFilteredList(@NamedArg("source") ObservableList<E> source, @NamedArg("predicate") Predicate<? super E> predicate) {
+        super(source);
+        filtered = new int[source.size() * 3 / 2 + 1];
+        if (predicate != null) {
+            setPredicate(predicate);
+        } else {
+            for (size = 0; size < source.size(); size++) {
+                filtered[size] = size;
+            }
+        }
+    }
+
+    public CustomFilteredList(@NamedArg("source") ObservableList<E> source) {
+        this(source, null);
+    }
+
+    public final ObjectProperty<Predicate<? super E>> predicateProperty() {
+        if (predicate == null) {
+            predicate = new ObjectPropertyBase<>() {
+                @Override
+                protected void invalidated() {
+                    refilter();
+                }
+
+                @Override
+                public Object getBean() {
+                    return CustomFilteredList.this;
+                }
+
+                @Override
+                public String getName() {
+                    return "predicate";
+                }
+            };
+        }
+        return predicate;
+    }
+
+    public final Predicate<? super E> getPredicate() {
+        return predicate == null ? null : predicate.get();
+    }
+
+    public final void setPredicate(Predicate<? super E> predicate) {
+        predicateProperty().set(predicate);
+    }
+
+    private Predicate<? super E> getPredicateImpl() {
+        if (getPredicate() != null) {
+            return getPredicate();
+        }
+        return t -> true;
+    }
+
+    @Override
+    protected void sourceChanged(Change<? extends E> c) {
+        beginChange();
+        while (c.next()) {
+            if (c.wasPermutated()) {
+                permutate(c);
+            } else if (c.wasUpdated()) {
+                update(c);
+            } else {
+                addRemove(c);
+            }
+        }
+        endChange();
+    }
+
+    /**
+     * Returns the number of elements in this list.
+     *
+     * @return the number of elements in this list
+     */
+    @Override
+    public int size() {
+        return size;
+    }
+
+    /**
+     * Returns the element at the specified position in this list.
+     *
+     * @param  index index of the element to return
+     * @return the element at the specified position in this list
+     * @throws IndexOutOfBoundsException {@inheritDoc}
```

Comment:
Can be removed. Either same text or just not adding more information 😅

```suggestion
```

Reply:
I just copied the class from `javafx.collections.transformation.FilteredList` and added the new method. 
I'll remove the java docs.